homeassistant/components/motionblinds_ble/config_flow.py

```diff
@@ -0,0 +1,211 @@
+    CONF_LOCAL_NAME,
+    CONF_MAC_CODE,
+    DOMAIN,
+    ERROR_COULD_NOT_FIND_MOTOR,
+    ERROR_INVALID_MAC_CODE,
+    ERROR_NO_BLUETOOTH_ADAPTER,
+    ERROR_NO_DEVICES_FOUND,
+    MotionBlindType,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema({vol.Required(CONF_MAC_CODE): str})
+
+
+class FlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for MotionBlinds BLE."""
+
+    def __init__(self) -> None:
+        """Initialize a ConfigFlow."""
+        self._discovery_info: BluetoothServiceInfoBleak | BLEDevice | None = None
+        self._mac_code: str | None = None
+        self._display_name: str | None = None
+        self._blind_type: MotionBlindType | None = None
+
+    async def async_step_bluetooth(
+        self, discovery_info: BluetoothServiceInfoBleak
+    ) -> FlowResult:
+        """Handle the bluetooth discovery step."""
+        _LOGGER.debug(
+            "Discovered MotionBlinds bluetooth device: %s", discovery_info.as_dict()
+        )
+        await self.async_set_unique_id(discovery_info.address)
+        self._abort_if_unique_id_configured()
+
+        self._discovery_info = discovery_info
+        self._mac_code = get_mac_from_local_name(discovery_info.name)
+        self._display_name = f"MotionBlind {self._mac_code}"
+        self.context["local_name"] = discovery_info.name
+        self.context["title_placeholders"] = {"name": self._display_name}
+
+        return await self.async_step_confirm()
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            mac_code = user_input[CONF_MAC_CODE]
+            # Discover with BLE
+            try:
+                await self.async_discover_motionblind(mac_code)
+            except NoBluetoothAdapter:
+                return self.async_abort(reason=EXCEPTION_MAP[NoBluetoothAdapter])
+            except NoDevicesFound:
+                return self.async_abort(reason=EXCEPTION_MAP[NoDevicesFound])
+            except tuple(EXCEPTION_MAP.keys()) as e:
+                errors = {
+                    "base": EXCEPTION_MAP[type(e)]
+                    if type(e) in EXCEPTION_MAP
+                    else str(type(e))
+                }
+                return self.async_show_form(
+                    step_id="user", data_schema=CONFIG_SCHEMA, errors=errors
+                )
+            return await self.async_step_confirm()
+
+        return self.async_show_form(
+            step_id="user", data_schema=CONFIG_SCHEMA, errors=errors
+        )
+
+    async def async_step_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm a single device."""
+        if user_input is not None:
+            self._blind_type = user_input[CONF_BLIND_TYPE]
+
+            assert self._discovery_info is not None
```

Comment:
Using `assert` in production code is bad style.

Reply:
This was added to make the tests and pre-commit pass. If I remove this, it says self._discovery_info can be None. I also tried adding a conditional that checks if it is None, but that results in code that cannot be tested thus not 100% code coverage of the config_flow. How should I solve this?

Reply:
Yeah, I had the exact same problem with a string, which I simply initialized as an empty string instead of `None`. You could do the same by not storing the `DiscoveryInfo` object but just the `discovery_info.name` and `discovery_info.address` as a quick workaround. Not sure how we are supposed to handle this though.

Reply:
I'll leave this open for now to see if anyone has better options of dealing with this

Reply:
You should use this:

```py
if TYPE_CHECKING:
    assert self._discovery_info is not None
```

At least I got told to handle it that way in my PR.

Reply:
Added that, thanks!